tests/test_strategy.py

```diff
@@ -63,10 +54,9 @@ class SignatureStrategyTestCase(unittest.TestCase):
     """
 
     @given(
-        strategies.text(
-            alphabet=strategies.sampled_from(_CODES),
-            min_size=1,
-            max_size=len(_CODES)
+        strategies.lists(
+            strategies.sampled_from(_CODES),
+            min_size=1, max_size=len(_CODES) - 1, unique=True
```

Comment:
Using `lists(..., unique=True).map(''.join)` instead of `text(...)` here is a subtle but important change - the `text` form is far less likely to try an almost-complete blacklist.  In fact, it never managed to generate the complete blacklist that was allowed before fixing the off-by-one error!